WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/CoinJoinTransactionArchiver.cs

```diff
@@ -50,6 +53,50 @@ public async Task<string> StoreJsonAsync(Transaction transaction, DateTimeOffset
 		return filePath;
 	}
 
+	public IEnumerable<TransactionInfo> ReadJson(DateTimeOffset from, DateTimeOffset to)
+	{
+		if (to < from)
+		{
+			throw new ArgumentException("Invalid time period", nameof(from));
+		}
+
+		List<string> dirsToRead = new();
+		int monthOffset = 0;
+
+		do
+		{
+			DateTimeOffset date = new(from.Year, from.Month + monthOffset, from.Day, 0, 0, 0, from.Offset);
+			if (date > to)
+			{
+				break;
+			}
+			dirsToRead.Add(Path.Combine(BaseStoragePath, $"{date:yyyy-MM}"));
+			monthOffset++;
+		}
+		while (true);
+
+		foreach (var montlyCoinJoinDirectory in dirsToRead.Select(d => new DirectoryInfo(d)).Where(di => di.Exists))
```

Comment:
I think, it would be safer to wrap this `foreach` in a lock (and also the part in `StoreJsonAsync` where we store a file) so that we cannot really read and write at the same time to avoid troubles.